lib/hammer_cli_foreman/defaults.rb

```diff
@@ -0,0 +1,29 @@
+require 'hammer_cli'
+module HammerCLIForeman
+  class Defaults < HammerCLI::BaseDefaultsProvider
+    SUPPORTED_DEFAULTS=[:organization_id, :location_id]
+    def self.support?(param)
+      SUPPORTED_DEFAULTS.any? {|s| s.to_s == param}
+    end
+
+    def self.supported_defaults
+      SUPPORTED_DEFAULTS
+    end
+
+    def self.get_defaults(param)
+      param = "default_organization" if param == :organization_id
+      param = "default_location" if param == :location_id
+      user = HammerCLIForeman.foreman_resource(:users).action(:index).call(:search =>"login="+HammerCLI::Settings.settings[:foreman][:username]) if HammerCLI::Settings.settings[:foreman][:username]
```

Comment:
To address the issue with missing username you can use HammerCLIForeman.credentials that ask interactively if necessary:
```ruby
      user = HammerCLIForeman.foreman_resource(:users).action(:index).call(:search =>"login="+HammerCLIForeman.credentials.username)
```